pkg/tracingpolicy/generictracingpolicy_test.go

```diff
@@ -527,3 +528,78 @@ func TestYamlNamespaced(t *testing.T) {
 	_, ok := tp.(TracingPolicyNamespaced)
 	require.True(t, ok)
 }
+
+func createTempFile(t *testing.T, data string) string {
+	file, err := os.CreateTemp(t.TempDir(), "tetragon-")
+	if err != nil {
+		t.Error(err)
+	}
+	path := file.Name()
+
+	_, err = file.WriteString(data)
+	if err != nil {
+		t.Error(err)
+	}
+	err = file.Close()
+	if err != nil {
+		t.Error(err)
+	}
+
+	t.Cleanup(func() {
+		err := os.Remove(path)
+		if err != nil {
+			t.Error(err)
+		}
+	})
```

Comment:
since you are using `file, err := os.CreateTemp(t.TempDir(), "tetragon-")`, you can remove the cleanup part since the `TempDir` folder [is already automatically removed](https://pkg.go.dev/testing#B.TempDir):
> TempDir returns a temporary directory for the test to use. The directory is automatically removed by Cleanup when the test and all its subtests complete.